security/cmd/istio_ca/main.go

```diff
@@ -142,13 +143,24 @@ func runCA() {
 	stopCh := make(chan struct{})
 	sc.Run(stopCh)
 
-	serviceController := controller.NewServiceController(cs.CoreV1(), opts.namespace,
-		registry.K8SServiceAdded, registry.K8SServiceDeleted, registry.K8SServiceUpdated)
-	serviceController.Run(stopCh)
-
 	if opts.grpcPort > 0 {
+		// start registry if gRPC server is to be started
+		reg := registry.GetIdentityRegistry()
+		ch := make(chan struct{})
+
+		// monitor service objects for `istioctl register` command
```

Comment:
Are we actually monitor the istioctl register command or monitor all services in K8s?

Reply:
service objects, in the same namespace as CA monitors. I'll change the wording.